File: algorithms/0349.intersection-of-two-arrays/solution.rb
# @param {Integer[]} nums1
# @param {Integer[]} nums2
# @return {Integer[]}
def intersection(nums1, nums2)
  # nums1 & nums2

  res = []
  nums1.uniq.each do |n|
    if nums2.include?(n)
      res << n
      nums2.delete(n)
    end
  end
  res
end

File: algorithms/0349.intersection-of-two-arrays/solution.rs
impl Solution {
    pub fn intersection(nums1: Vec<i32>, nums2: Vec<i32>) -> Vec<i32> {
        use std::collections::HashSet;
        let set1: HashSet<i32> = nums1.into_iter().collect();
        let set2: HashSet<i32> = nums2.into_iter().collect();

        set1.intersection(&set2).cloned().collect()
    }
}
